packages/core/src/components/SupportButton/SupportButton.tsx

```diff
@@ -78,13 +76,45 @@ const SupportListItem = ({ item }: { item: SupportItem }) => {
   );
 };
 
-export const SupportButton = ({ children }: PropsWithChildren<Props>) => {
-  const { items } = useSupportConfig();
+const createDisplayedItems = (
+  showCondition: Boolean | undefined,
+  generalItems: SupportItem[],
+  supportItems: SupportItem[] | undefined,
+) => {
+  if (showCondition && supportItems) {
+    return supportItems;
+  }
+  if (supportItems) {
+    return [...supportItems, ...generalItems];
+  }
+  return generalItems;
+};
+
+type Props = {};
 
+type SupportButtonProps = {
+  title?: string;
+  supportItems?: SupportItem[];
+  hideDefaultItems?: boolean;
+};
+
+export const SupportButton = ({
+  supportItems,
+  title,
+  hideDefaultItems = false,
+  children,
+}: SupportButtonProps & PropsWithChildren<Props>) => {
```

Comment:
this should have been `PropsWithChildren<SupportButtonProps>`, but we usually don't even do that - just add `children?: React.ReactNode` as a field directly in `SupportButtonProps` instead. And delete `Props`